src/renderer/canvas/node.ts

```diff
@@ -156,34 +157,34 @@ const setupShadow = <N extends INodeBase, E extends IEdgeBase>(
   context: CanvasRenderingContext2D,
   node: INode<N, E>,
 ) => {
-  if (node.style.shadowColor) {
-    context.shadowColor = node.style.shadowColor.toString();
+  if (node.getStyle().shadowColor) {
+    context.shadowColor = (node.getStyle().shadowColor as string | Color).toString();
   }
-  if (node.style.shadowSize) {
-    context.shadowBlur = node.style.shadowSize;
+  if (node.getStyle().shadowSize) {
+    context.shadowBlur = node.getStyle().shadowSize as number;
   }
-  if (node.style.shadowOffsetX) {
-    context.shadowOffsetX = node.style.shadowOffsetX;
+  if (node.getStyle().shadowOffsetX) {
+    context.shadowOffsetX = node.getStyle().shadowOffsetX as number;
   }
-  if (node.style.shadowOffsetY) {
-    context.shadowOffsetY = node.style.shadowOffsetY;
+  if (node.getStyle().shadowOffsetY) {
+    context.shadowOffsetY = node.getStyle().shadowOffsetY as number;
   }
```

Comment:
In these cases, it would be better to get style before checking each condition because right now you are doing 8 deep copies from your node style (that has 30-40 properties). On N number of nodes and render cycles, this will slow down things a lot. This applies to all other usages like this.